### PR TITLE
Refs #36688 - Option to use wget for host registration

### PR DESCRIPTION
**Pull request by goarsna**
Documents https://github.com/theforeman/foreman/pull/9808

As documentation for https://github.com/theforeman/foreman/pull/9808 was prematurely merged in https://github.com/theforeman/foreman-documentation/pull/2376 and reverted in https://github.com/theforeman/foreman-documentation/pull/2793, @Lennonka kindly created https://github.com/theforeman/foreman-documentation/pull/2914 to re-add documentation for https://github.com/theforeman/foreman/pull/9808 once it is merged. Additionally I created https://github.com/theforeman/foreman-documentation/pull/3163 as follow-up PR to https://github.com/theforeman/foreman-documentation/pull/2914.

Due to heavy rewriting of the module, we decided to close https://github.com/theforeman/foreman-documentation/pull/2914 as well as https://github.com/theforeman/foreman-documentation/pull/3163 and to open this new PR for documenting https://github.com/theforeman/foreman/pull/9808 instead.

* [x] I am familiar with the [contributing](https://github.com/theforeman/foreman-documentation/blob/master/CONTRIBUTING.md) guidelines.

Please cherry-pick my commits into:

* [ ] Foreman 3.12/Katello 4.14 (Satellite 6.16)
* [ ] Foreman 3.11/Katello 4.13
* [ ] Foreman 3.10/Katello 4.12
* [ ] Foreman 3.9/Katello 4.11 (Satellite 6.15; orcharhino 6.8/6.9/6.10)
* [ ] Foreman 3.8/Katello 4.10
* [ ] Foreman 3.7/Katello 4.9 (Satellite 6.14)
* [ ] Foreman 3.6/Katello 4.8
* [ ] Foreman 3.5/Katello 4.7 (Satellite 6.13; orcharhino 6.6/6.7)
* We do not accept PRs for Foreman older than 3.5.

### CODE REVIEW

File: guides/common/modules/proc_configuring-a-host-for-registration.adoc [hunk 1]
```diff
@@ -7,6 +7,7 @@ You can use a configuration management tool to configure multiple hosts at once.
 .Prerequisites
 * The host must be using a supported operating system.
 For more information, see xref:supported-clients-in-registration_{context}[].
+* Either `curl` or `wget` must be installed on the host.
```
Comment (Lennonka) [inline guides/common/modules/proc_configuring-a-host-for-registration.adoc:10]: I don't think it's needed in this procedure, I'd move it to Prerequisites in _Registering a host_.

File: guides/common/modules/proc_registering-a-host.adoc [hunk 2]
```diff
@@ -42,40 +42,42 @@ endif::[]
...
+If an attacker, located in the network between {Project} and your host, fetches the CA file from the first insecure call, the attacker will be able to access the content of the API calls to and from your host and the JSON Web Tokens (JWT).
+Therefore, if you have chosen to deploy SSH keys during registration, the attacker will be able to access your host using the SSH key.
 * On the *Advanced* tab, in the *Repositories* field, you can list repositories to be added before the registration is performed.
 You do not have to specify repositories if you provide them in an activation key.
 * On the *Advanced* tab, in the *Token lifetime (hours)* field, you can change the validity duration of the JSON Web Token (JWT) that {Project} uses for authentication.
-The duration of this token defines how long the generated `curl` command works.
+The duration of this token defines how long the generated registration command works.
 +
-Note that {Project} applies the permissions of the user who generates the `curl` command to authorization of hosts.
+Note that {Project} applies the permissions of the user who generates the regiastration command to authorization of your host.
```
Comment (Lennonka) [inline guides/common/modules/proc_registering-a-host.adoc:72]:
```suggestion
Note that {Project} applies the permissions of the user who generates the registration command to authorization of your host.
```
Typo

File: guides/common/modules/ref_registration-methods.adoc [hunk 3]
```diff
@@ -4,7 +4,7 @@
 You can use the following methods to register hosts to {Project}:
 
 Global registration::
-You generate a `curl` command from {Project} and run this command from an unlimited number of hosts to register them using provisioning templates over the {Project} API.
+You generate a `curl` or `wget` command from {Project} and run this command from an unlimited number of hosts to register them using provisioning templates over the {Project} API.
```
Comment (Lennonka) [inline guides/common/modules/ref_registration-methods.adoc:7]:
```suggestion
You generate a registration command from {Project} and run this command on an unlimited number of hosts to register them by using provisioning templates over the {Project} API.
```
I'd use the generic term (fewer details) in this context. We elaborate on the command in the specific section.

**Review by Lennonka — CHANGES_REQUESTED** [review]
Overally looks good, just a couple of details

**Comment by goarsna** [comment]
Thanks @Lennonka for your review. Applied all suggestions and rebased. :)

**Comment by goarsna** [comment]
Just fyi: https://github.com/theforeman/foreman/pull/9808 has been merged two days ago.

**Review by maximiliankolb — APPROVED** [review]
Thanks Goars, LGTM.

File: guides/common/modules/proc_registering-a-host.adoc [hunk 4]
```diff
@@ -42,40 +43,42 @@ endif::[]
...
+Note that {Project} applies the permissions of the user who generates the registration command to authorization of your host.
 If the user loses or gains additional permissions, the permissions of the JWT change too.
 Therefore, do not delete, block, or change permissions of the user during the token duration.
 +
 The scope of the JWTs is limited to the registration endpoints only and cannot be used anywhere else.
 
 .CLI procedure
-. Use the `hammer host-registration generate-command` to generate the `curl` command to register the host.
-. On the host that you want to register, run the `curl` command as `root`.
+. Use the `hammer host-registration generate-command` to generate the `curl` command to register your host.
```
Comment (nadjaheitmann) [inline guides/common/modules/proc_registering-a-host.adoc:80]: What about CLI? Is this working for hammer as well? Or is wget planned to be implemented later?

File: guides/common/modules/proc_registering-a-host.adoc [hunk 5]
```diff
@@ -42,40 +43,42 @@ endif::[]
...
 . Click *Generate* to generate a `curl` command.
 . Run the `curl` command as `root` on the host that you want to register.
 After registration completes, any Ansible roles assigned to a host group you specified when configuring the registration template will run on the host.
 
 The registration details that you can specify include the following:
 
-* On the *General* tab, in the *{SmartProxy}* field, you can select the {SmartProxy} to register hosts through.
+* On the *General* tab, in the *{SmartProxy}* field, you can select the {SmartProxy} to register your host through.
 A {SmartProxy} behind a load balancer takes precedence over a {SmartProxy} selected in the {ProjectWebUI} as the content source of the host.
+* On the *General* tab, in the *Download utility* field, you can select `wget` to register your host using a `wget` command.
```
Comment (nadjaheitmann) [inline guides/common/modules/proc_registering-a-host.adoc:58]:
```suggestion
* On the *General* tab, in the *Download utility* field, you can select `wget` if you want to register your host using a `wget` command.
```

File: guides/common/modules/proc_registering-a-host.adoc (hunk 4 above)
Comment (nadjaheitmann) [reply guides/common/modules/proc_registering-a-host.adoc:80]: Just found out myself: 
image

And I tried it, it actually does generate a command. So please add this to the documentation.

File: guides/common/modules/proc_registering-a-host.adoc (hunk 5 above)
Comment (Lennonka) [reply guides/common/modules/proc_registering-a-host.adoc:58]: + `by using`

File: guides/common/modules/proc_registering-a-host.adoc (hunk 4 above)
Comment (Lennonka) [reply guides/common/modules/proc_registering-a-host.adoc:80]: I don't think it's necessary to mention the option specifically. They can see it in Hammer help as said below.

Comment (nadjaheitmann) [reply guides/common/modules/proc_registering-a-host.adoc:80]: The sentence above only mentions the curl command but not wget at all. That is what I stumbled across.

Comment (Lennonka) [reply guides/common/modules/proc_registering-a-host.adoc:80]: Oh, I see! In that case I suggest:
```suggestion
. Use the `hammer host-registration generate-command` to generate the registration command to register your host.
```

**Review by Lennonka — APPROVED** [review]
LGTM

File: guides/common/modules/proc_registering-a-host.adoc (hunk 4 above)
Comment (goarsna) [reply guides/common/modules/proc_registering-a-host.adoc]: I applied the suggested changes for the CLI procedure. The API procedure only refers to the API reference. The `download_utility` is documented there.

Regarding the Ansible procedure: AFAIK for now, the usage of `download_utility` has to be documented in [plugins/modules/registration_command.py in foreman-ansible-modules](https://github.com/theforeman/foreman-ansible-modules/blob/develop/plugins/modules/registration_command.py). But I'll have to take another look and would open a PR accordingly.

**Review by nadjaheitmann — APPROVED** [review]

**Comment by goarsna** [comment]
Thanks a lot for your reviews @Lennonka, @maximiliankolb , and @nadjaheitmann!

**Comment by maximiliankolb** [comment]
Tech review done by Nadja. :heavy_check_mark:

**Review by maximiliankolb — APPROVED** [review]
Thanks Goars, LGTM.

**Pull request merged**